### PR TITLE
SD-1453 As a Storefront API user I can set User MetaData when creating an Account

### PR DESCRIPTION
**Pull request by aleksandarpetrushev**
https://spark-solutions.atlassian.net/browse/SD-1453
https://api.spreecommerce.org/docs/api-v2/b3A6MzE0MjczNQ-create-an-account

### CODE REVIEW

File: api/app/serializers/spree/v2/storefront/user_serializer.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ module Storefront
       class UserSerializer < BaseSerializer
         set_type :user
 
-        attributes :email
+        attributes :email, :public_metadata, :private_metadata
```
Comment (damianlegawiec) [inline api/app/serializers/spree/v2/storefront/user_serializer.rb:7]: @aleksandarpetrushev we don't want to expose `private_metadata` only public, however, we need a way to store private metadata via storefront api

File: api/docs/v2/storefront/index.yaml [hunk 2]
```diff
@@ -63,6 +63,16 @@ paths:
...
+                    public_metadata:
+                      type: object
+                      example:
+                        number_of_orders: 0
+                      description: The public metadata for this User
+                    private_metadata:
+                      type: object
+                      example:
+                        has_abandoned_cart: false
+                      description: The private metadata for this User
```
Comment (damianlegawiec) [inline api/docs/v2/storefront/index.yaml:75]:
```suggestion
```

**Review by damianlegawiec — CHANGES_REQUESTED** [review]

File: api/docs/v2/storefront/index.yaml (hunk 2 above)
Comment (aleksandarpetrushev) [reply api/docs/v2/storefront/index.yaml:75]: This is for the parameter when creating the Account. You set the private metadata by passing the private_metadata param. Then in the response I have changed it to only show the public metadata

File: api/docs/v2/storefront/index.yaml [hunk 3]
```diff
@@ -63,6 +63,16 @@ paths:
                     password_confirmation:
                       type: string
                       example: spree123
+                    public_metadata:
+                      type: object
+                      example:
+                        number_of_orders: 0
```
Comment (damianlegawiec) [inline api/docs/v2/storefront/index.yaml:69]:
```suggestion
                        user_segment: 'supplier'
```

Let's use this example as number of orders is returned in `completed_orders` attribute

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**